src/helpers/similarity.ts

```diff
@@ -0,0 +1,86 @@
+import { getBotContext } from "../bindings";
+import { Payload, Choices } from "../types";
+import { listAllIssuesForRepo, upsertCommentToIssue } from "../helpers/issue";
+import axios from "axios";
+
+export const findDuplicateOne = async () => {
+  const { payload: _payload } = getBotContext();
+  const issue = (_payload as Payload).issue;
+  const issues = await listAllIssuesForRepo("all");
+  issues.shift();
+  const importantWords = await extractImportantWords();
+  const wordCount = importantWords.length;
+  for (const iss of issues) {
+    if (iss.body) {
+      const probability = (countIncludedWords(iss.body, importantWords) * 100.0) / wordCount;
+      if (probability > parseInt(process.env.SIMILARITY_THRESHOLD || "80")) {
+        if (issue?.number) {
+          await upsertCommentToIssue(issue?.number, `Similar issue (${iss.title}) found at ${iss.html_url}.\nSimilarity is about ${probability}%`, "created");
+          return;
+        }
+      }
+    }
+  }
+};
+
+const countIncludedWords = (inputString: string, words: string[]): number => {
```

Comment:
The input string is in Markdown syntax so you should preprocess the text by removing URLs, images, ... and put all the words in an array/set, so you can compare with important words.

Reply:
But this function only counts words, ChatGPT is not involved here

Reply:
Okay. Is it okay too use commentParser?

Reply:
Yes, you can use [parseComments](https://github.com/ubiquity/ubiquibot/blob/dfb48b101d50c8134a51221384968f39fb7930b9/src/helpers/comment.ts#L22C23-L22C23)

Reply:
I think parsing every issues will consume some more time. @whilefoo

Reply:
Changed strategy and measure similarity using ChatGPT. I think this is resolved.